File: design_patterns_study/230407_factory/java/abstractFactory/Main.java
package abstractFactory;

public class Main{
    public static void main(String[] args) {

        System.out.println("Hello World");

        System.out.println("== From calling classes directly ====================================");
        BlueTriangle tri = new BlueTriangle();
        tri.get();

        BlueRectangle rec = new BlueRectangle();
        rec.get();

        BluePentagon pen = new BluePentagon();
        pen.get();


        RedTriangle tri2 = new RedTriangle();
        tri2.get();

        RedRectangle rec2 = new RedRectangle();
        rec2.get();

        RedPentagon pen2 = new RedPentagon();
        pen2.get();


        GreenTriangle tri3 = new GreenTriangle();
        tri3.get();

        GreenRectangle rec3 = new GreenRectangle();
        rec3.get();

        GreenPentagon pen3 = new GreenPentagon();
        pen3.get();

        System.out.println("== From calling the abstractFactory ====================================");

        Polygon tri4 = MetaFactoryPolygon.create_factory("blue").create(3);
        tri4.get();

        Polygon rec4 = MetaFactoryPolygon.create_factory("blue").create(4);
        rec4.get();

        Polygon pen4 = MetaFactoryPolygon.create_factory("blue").create(5);
        pen4.get();


        Polygon tri5 = MetaFactoryPolygon.create_factory("red").create(3);
        tri5.get();

        Polygon rec5 = MetaFactoryPolygon.create_factory("red").create(4);
        rec5.get();

        Polygon pen5 = MetaFactoryPolygon.create_factory("red").create(5);
        pen5.get();


        Polygon tri6 = MetaFactoryPolygon.create_factory("green").create(3);
        tri6.get();

        Polygon rec6 = MetaFactoryPolygon.create_factory("green").create(4);
        rec6.get();

        Polygon pen6 = MetaFactoryPolygon.create_factory("green").create(5);
        pen6.get();
    }
}

File: design_patterns_study/230407_factory/go/abstractFactory/main.go
package main

import (
	"abstractfactory/factory"
	"abstractfactory/polygon"
	"fmt"
)

func main() {
	fmt.Println("Hello World")
	fmt.Println("== From calling classes directly ====================================")

	tri := polygon.BlueTriangle{}
	tri.Get()
	tri2 := polygon.RedTriangle{}
	tri2.Get()
	tri3 := polygon.GreenTriangle{}
	tri3.Get()

	rec := polygon.BlueRectangle{}
	rec.Get()
	rec2 := polygon.RedRectangle{}
	rec2.Get()
	rec3 := polygon.GreenRectangle{}
	rec3.Get()

	pen := polygon.BluePentagon{}
	pen.Get()
	pen2 := polygon.RedPentagon{}
	pen2.Get()
	pen3 := polygon.GreenPentagon{}
	pen3.Get()

	fmt.Println("== From calling the factoryMethod ====================================")

	factory.AbstractFactoryPolygon{}.CreateFactory("blue").Create(3).Get()
	factory.AbstractFactoryPolygon{}.CreateFactory("blue").Create(4).Get()
	factory.AbstractFactoryPolygon{}.CreateFactory("blue").Create(5).Get()

	factory.AbstractFactoryPolygon{}.CreateFactory("red").Create(3).Get()
	factory.AbstractFactoryPolygon{}.CreateFactory("red").Create(4).Get()
	factory.AbstractFactoryPolygon{}.CreateFactory("red").Create(5).Get()

	factory.AbstractFactoryPolygon{}.CreateFactory("green").Create(3).Get()
	factory.AbstractFactoryPolygon{}.CreateFactory("green").Create(4).Get()
	factory.AbstractFactoryPolygon{}.CreateFactory("green").Create(5).Get()

}
